### PR TITLE
Added *.cabal to gitignore and removed .cabal files

### LINKED ISSUE #860: cabal files not being autogenerated by stack due to hpack versioning
Because we are committing cabal files into stack, they have a version of hpack that is not mine encoded against them. I'm not even sure how I can upgrade to a later version of hpack; I have a version of stack *built against master* and I still don't have up-to-date hpack (maybe I have to force the resolver for which stack builds itself)

In any case this problem can be dodged entirely by *not committing the cabal files to the git repository*. Since they are autogenerated, there is no reason for them to be committed and doing so risks skew between the generated cabal files and the package.yaml.

Thoughts? If theres no large disagreement I'd like to just purge all of the cabal files from the repository.

### PR DESCRIPTION
**Pull request by noughtmare**
Fixes #860

### CODE REVIEW

**Comment by noughtmare** [comment]
I guess the tests are failing because hpack is not installed... (I don't know how travis works)

**Comment by jhance** [comment]
I'll take a look

File: yi-language/package.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+version: '0.2.1'
+synopsis: Collection of language-related Yi libraries.
+maintainer: Dmitry Ivanov <ethercrow@gmail.com>
+license: GPL-2
+github: yi-editor/yi
+category: Yi
+
+ghc-options: -Wall -ferror-spans
+
+#dependencies:
```
Comment (jhance) [inline yi-language/package.yaml]: If I uncomment this part and remove the dependencies in the `library` part, I generate a correct cabal file.

**Comment by noughtmare** [comment]
Fixed

**Comment by jhance** [comment]
I had the problem in *both* PRs.

**Pull request merged**